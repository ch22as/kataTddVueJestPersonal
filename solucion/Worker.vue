<template>
  <div class="container worker">

      <input type="text" placeholder="Introduzca texto" name="" >

      <button @click="send" >Enviar</button>



  </div>
</template>

<script>
import { mapActions } from 'vuex'

export default {
    name: 'Worker',
    methods: {
        ...mapActions({
            send: 'FETCH'
        })
    }
}
</script>

<style>
.worker{
    margin:1%;
    padding:1%;
    background: black;
}

</style>