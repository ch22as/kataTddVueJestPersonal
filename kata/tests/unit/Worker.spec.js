import { mount, createLocalVue } from '@vue/test-utils';
import Worker from '@/components/Worker';   
import Vuex from 'vuex';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('Worker', () =>{
    
    let store, actions;

    beforeEach(() =>{
        actions = { FECH: jest.fn() };
        
        store = new Vuex.Store({ actions })
    });

    const build = () => {
        const wrapper = mount(Worker, { store, localVue });
    
        return {
            wrapper,
            input: () => wrapper.find('input'),
            button: () => wrapper.find('button')
        }
    };


    it('Should render it', () =>{
        const { wrapper } = build();

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('input', () =>{
        const { input } = build();

        expect(input().exists()).toBe(true);
        expect(input().attributes().type).toEqual("text");
        expect(input().attributes().placeholder).toEqual("Introduzca texto")
    });

    it('button', () =>{
        const { button } = build();

        expect(button().exists()).toBe(true);
        expect(button().text()).toEqual("Enviar");
        
    });

    it.skip('button click', () =>{
        const { button } = build();

        button().trigger('click');
        expect(actions.FECH).toHaveBeenCalled();
    })

})