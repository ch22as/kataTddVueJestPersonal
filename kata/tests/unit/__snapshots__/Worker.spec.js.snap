// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Worker Should render it 1`] = `<div class="container worker"><input type="text" placeholder="Introduzca texto" name=""> <button>Enviar</button></div>`;
